policy.go

```diff
@@ -26,11 +26,14 @@ const (
 func calcMinRequiredTxRelayFee(serializedSize int64, minRelayTxFee btcutil.Amount) int64 {
 	// Calculate the minimum fee for a transaction to be allowed into the
 	// mempool and relayed by scaling the base fee (which is the minimum
-	// free transaction relay fee).  minRelayTxFee is in Satoshi/KB, so
-	// divide the transaction size by 1000 to convert to kilobytes.  Also,
-	// integer division is used so fees only increase on full kilobyte
-	// boundaries.
-	minFee := (1 + serializedSize/1000) * int64(minRelayTxFee)
+	// free transaction relay fee). minTxRelayFee is in Satoshi/KB so
+	// multiply by serializedSize (which is in KB) and divide by 1000 to get
+	// minimum Satoshis.
+	minFee := int64(minRelayTxFee)
+	// If we have more than 1KB calculate the new minFee
+	if serializedSize > 1000 {
```

Comment:
More examples of tx sizes and fees (I'd suggest adding these to the tests):

tx size: 1500 bytes
minRelayTxFee: 5000
minFee: 7500

tx size: 1500 bytes
minRelayTxFee: 3000
minFee: 4500

tx size: 782 bytes
minRelayTxFee: 5000
minFee: 3910

tx size: 782 bytes
minRelayTxFee: 3000
minFee: 2346

tx size: 782 bytes
minRelayTxFee: 2550
minFee: 1994